app/src/main/java/io/pumpkinz/pumpkinreader/service/CommentSource.java

```diff
@@ -0,0 +1,124 @@
+package io.pumpkinz.pumpkinreader.service;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+
+import io.pumpkinz.pumpkinreader.model.Comment;
+import io.pumpkinz.pumpkinreader.model.News;
+import rx.Observable;
+import rx.functions.Func1;
+
+
+public class CommentSource {
+
+    private static CommentSource instance;
+
+    private CommentSource() {
+    }
+
+    public static CommentSource service() {
```

Comment:
I'm wondering if there's better pattern than using singleton. I'm starting to think I'm over-using singleton, here.